### ISSUE TITLE
[Network] Method "getNodeAt" is not working at click on selected node

### ISSUE THREAD

**Issue by Alexthetiger** [almende/vis#1847]
Method "getNodeAt" is not working properly inside "click" event at click on selected node. Thereby I haven't information about click target. Example: I click on selected node and in "click" event callback I call "this.visNetwork.getNodeAt(obj.pointer.DOM)" it returns undefined.

**Comment by spurdy** [comment]
Looks like the 'unselect' and 'select' actions that fire before the getNodeAt method would be called removes the node.shape.width and node.shape.height which causes issues for a method that getNodeAt() calls. You can get around the issue by calling a redraw(). Here's a fiddle that shows both the problem and the work-around:
[JS Fiddle](https://jsfiddle.net/qwhzofqq/1/)

**Comment by Alexthetiger** [comment]
Yes, it works, thanks.

**Comment by wimrijnders** [comment]
It should be noted here that the parameter of the callback in `network.on('click', function(obj){...})` also contains the node id clicked on.

```
network.on('click', function(obj){ 
  var nodeId = this.getNodeAt(obj.pointer.DOM);   // returns 'undefined' due to bug

  nodeId = obj.nodes[0];      // the node id that getNodeAt() should be returning
}); 
```

This is, in my opinion, the preferred solution to getting the node id within an event listener. It will work with almost all event listeners; an exception is `showPopup`, which returns the node id directly in parameter `obj`.

**Comment by bradh** [comment]
Is this now fixed pending release?

**Comment by wimrijnders** [comment]
@bradh Yes, this is fixed by #2885.
